cpp/oneapi/dal/algo/pca/backend/cpu/train_kernel_svd.cpp

```diff
@@ -14,17 +14,101 @@
+
 #include "oneapi/dal/algo/pca/backend/cpu/train_kernel.hpp"
+#include "oneapi/dal/backend/interop/common.hpp"
+#include "oneapi/dal/backend/interop/error_converter.hpp"
+#include "oneapi/dal/backend/interop/table_conversion.hpp"
+#include "oneapi/dal/table/row_accessor.hpp"
 
 namespace oneapi::dal::pca::backend {
 
+using std::int64_t;
+using dal::backend::context_cpu;
+
+namespace daal_pca = daal::algorithms::pca;
+namespace interop = dal::backend::interop;
+
+template <typename Float, daal::CpuType Cpu>
+using daal_pca_svd_kernel_t = daal_pca::internal::
+    PCASVDBatchKernel<Float, daal_pca::interface3::BatchParameter<Float, daal_pca::svdDense>, Cpu>;
+
+template <typename Float, typename Task>
+static train_result<Task> call_daal_kernel(const context_cpu& ctx,
+                                           const descriptor_base<Task>& desc,
+                                           const table& data) {
+    const int64_t row_count = data.get_row_count();
+    const int64_t column_count = data.get_column_count();
+    const int64_t component_count = desc.get_component_count();
+
+    auto arr_data = row_accessor<const Float>{ data }.pull();
+    auto arr_eigvec = array<Float>::empty(column_count * component_count);
+    auto arr_eigval = array<Float>::empty(1 * component_count);
+    auto arr_means = array<Float>::empty(1 * component_count);
+    auto arr_vars = array<Float>::empty(1 * component_count);
+
+    // TODO: read-only access performed with deep copy of data since daal numeric tables are mutable.
+    // Need to create special immutable homogen table on daal interop side
+
+    // TODO: data is table, not a homogen_table. Think better about accessor - is it enough to have just a row_accessor?
+    const auto daal_data =
+        interop::convert_to_daal_homogen_table(arr_data, row_count, column_count);
+    const auto daal_eigenvectors =
+        interop::convert_to_daal_homogen_table(arr_eigvec, column_count, component_count);
+    const auto daal_eigenvalues =
+        interop::convert_to_daal_homogen_table(arr_eigval, 1, component_count);
+    const auto daal_means = interop::convert_to_daal_homogen_table(arr_means, 1, component_count);
+    const auto daal_variances =
+        interop::convert_to_daal_homogen_table(arr_vars, 1, component_count);
+
+    daal_pca::internal::InputDataType dtype = daal_pca::internal::nonNormalizedDataset;
+
+    daal_pca::interface3::BatchParameter<Float, daal_pca::svdDense> parameter;
+
+    auto normalizationAlgorithm = parameter.normalization;
+    normalizationAlgorithm->input.set(daal::algorithms::normalization::zscore::data, daal_data);
+    auto algParameter = &(normalizationAlgorithm->parameter());
+    algParameter->resultsToCompute |= daal::algorithms::normalization::zscore::mean;
+    algParameter->resultsToCompute |= daal::algorithms::normalization::zscore::variance;
+
+    interop::status_to_exception(dal::backend::dispatch_by_cpu(ctx, [&](auto cpu) {
```

Comment:
I guess, `interop::call_daal_kernel` (see [here](https://github.com/oneapi-src/oneDAL/blob/b39ee10d315abf480f2efbf209c27e3e14987ae9/cpp/oneapi/dal/backend/interop/common.hpp#L47)) can be used instead

Reply:
need to be discussed

Reply:
fixed